### PR TITLE
[infra] Introduce onert-micro unit-tests workflow

### PR DESCRIPTION
**Pull request by SlavikMIPT**
This commit introduce onert-micro unit-tests workflow.

ONE-DCO-1.0-Signed-off-by: Vyacheslav Bazhenov <slavikmipt@gmail.com>

### CODE REVIEW

**Comment by SlavikMIPT** [comment]
@Torrero

**Comment by hseok-oh** [comment]
`ubuntu-20.04` runner is github-hosted runner. Is it enough spec to build and test `onert-micro` ?

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources

cc/ @chunseoklee

**Comment by chunseoklee** [comment]
> `ubuntu-20.04` runner is github-hosted runner. Is it enough spec to build and test `onert-micro` ?
> 
> https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
> 
> cc/ @chunseoklee 

I heard that @SlavikMIPT already tested on forked repo.

**Comment by SlavikMIPT** [comment]
> `ubuntu-20.04` runner is github-hosted runner. Is it enough spec to build and test `onert-micro` ?
> 
> https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources
> 
> cc/ @chunseoklee

yes

File: .github/workflows/run-onert-micro-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    # Skip on draft, check on draft -> ready
+    if: github.event.pull_request.draft == false
+
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            onert-micro:
+              - 'onert-micro/**'
```
Comment (hseok-oh) [inline .github/workflows/run-onert-micro-unit-tests.yml:31]: IMO, it's better to add this workflow file itself to filter.

**Comment by hseok-oh** [comment]
Workflow looks good.

I checked forked repo's test run: https://github.com/SlavikMIPT/ONE/actions/runs/4992675265
It seems it installs `arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-eabi` to build `onert-micro`, but build log shows it uses gcc 9.4.0

```
-- The C compiler identification is GNU 9.4.0
-- The CXX compiler identification is GNU 9.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
```

IMHO, it uses native gcc build tools, not cross build tools.
Is it intend build test? or my misreading?

**Comment by SlavikMIPT** [comment]
> Workflow looks good.
> 
> I checked forked repo's test run: https://github.com/SlavikMIPT/ONE/actions/runs/4992675265 It seems it installs `arm-gnu-toolchain-12.2.rel1-x86_64-arm-none-eabi` to build `onert-micro`, but build log shows it uses gcc 9.4.0
> 
> ```
> -- The C compiler identification is GNU 9.4.0
> -- The CXX compiler identification is GNU 9.4.0
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working C compiler: /usr/bin/cc - skipped
> ```
> 
> IMHO, it uses native gcc build tools, not cross build tools. Is it intend build test? or my misreading?

1. This is the main compiler for onert-micro and its presence is checked in https://github.com/Samsung/ONE/blob/8427211fcbee6ef2acee5b7a96194c13f4d4b6e3/onert-micro/CMakeLists.txt#L8
We build with native gcc tools only for running unit tests on CI (because the issue with the simulator/device farm is not solved yet)
2. I am going to also add build check with arm-none-eabi in next PR

**Review by hseok-oh — APPROVED** [review]
LGTM

File: .github/workflows/run-onert-micro-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+jobs:
+  run-onert-micro-unit-tests:
+    name: Run onert-micro Unit tests
+    runs-on: ubuntu-20.04
+    # Skip on draft, check on draft -> ready
+    if: github.event.pull_request.draft == false
+
+    steps:
+      - uses: dorny/paths-filter@v2
```
Comment (hseok-oh) [inline .github/workflows/run-onert-micro-unit-tests.yml:26]: @SlavikMIPT Sorry. I have one additional question. Why don't you use built-in filter? 
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

Comment (SlavikMIPT) [reply .github/workflows/run-onert-micro-unit-tests.yml:26]: Thanks - I will refactor this, I don't have much experience with github-actions

**Review by hseok-oh — APPROVED** [review]

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**